### PR TITLE
Added unary plus for Complex{Bool}

### PR DESCRIPTION
**Pull request by eulerkochy**
This is in reference to #30749 , I have created an overloading unary plus operator. 

`+(Complex(true))` results in 1+0im , and `+(Complex(false))` results in 0+0im

### CODE REVIEW

File: base/complex.jl [hunk 1]
```diff
@@ -8,6 +8,9 @@ Complex number type with real and imaginary part of type `T`.
 `ComplexF16`, `ComplexF32` and `ComplexF64` are aliases for
 `Complex{Float16}`, `Complex{Float32}` and `Complex{Float64}` respectively.
 """
+
+import Base.+
```
Comment (StefanKarpinski) [inline base/complex.jl]: You're in Base so you don't need this.

File: base/complex.jl [hunk 2]
```diff
@@ -272,6 +275,10 @@ muladd(z::Complex, w::Complex, x::Complex) =
     Complex(muladd(real(z), real(w), real(x)) - imag(z)*imag(w), # TODO: use mulsub given #15985
             muladd(real(z), imag(w), muladd(imag(z), real(w), imag(x))))
 
+#unary + handling Complex{Bool}
+
++(z::Complex{Bool}) = Complex(convert(Int,z))
```
Comment (StefanKarpinski) [inline base/complex.jl]: A nice way to write this would be `Complex(+z.re, +z.im)` so that it just delegates to whatever real unary `+` does.

**Comment by StefanKarpinski** [comment]
Thanks for the contribution! I've left a couple of comments. It would also be great to have a test or two for this and an entry in the NEWS file. This qualifies as a "minor change" which should be noted there.

File: base/complex.jl (hunk 1 above)
Comment (eulerkochy) [reply base/complex.jl]: My bad! :(

File: base/complex.jl (hunk 2 above)
Comment (eulerkochy) [reply base/complex.jl]: Ok!

**Comment by eulerkochy** [comment]
Re-opening this

**Comment by eulerkochy** [comment]
.

**Pull request closed without merging**